### DISCUSSION TITLE
How to accept sending mail only to some domains ?

### DISCUSSION THREAD

**Discussion by Maloupi** [emersion/maddy#630] [Q&A] [answered]
I want to forbid sending mail to all domains except one. How can i do this ?
I see the "[Email with domain](https://maddy.email/reference/table/email_with_domains/)" option but i did not understood how it works. 
My idea was that if an account send a mail to user@baddomain.com it will be sent to user@gooddomain.com but i'm not even sure it works like this. It's not exactly what i need but if this is not possible to accept only a domain, it may be usable in my case.

**Comment by foxcpp** [comment, marked as answer]
You can use `destination` directives in `submission` block to define routing for each domain separately. In default config,, `default_destination` sends outbound messages via queue. You can change it to reject and then move queue use to `destination gooddomain.com` to allow such delivery for this destination domain. Like this:

```
submission tls://0.0.0.0:465 tcp://0.0.0.0:587 {
    ...
    source $(local_domains) {
        ...

        destination postmaster $(local_domains) {
            deliver_to &local_routing
        }
        destination gooddomain.com {
            modify {
                dkim $(primary_domain) $(local_domains) default
            }
            deliver_to &remote_queue
        }
        default_destination {
            reject 501 5.1.8 "Not allowed domains"
        }
    }
    ...
}
```

**Reply by Maloupi** [reply]
Thank you, i better understand how it works.
Great, it works very well.
